packages/yargs/lib/request.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+const morgan = require('morgan');
+const split = require('split');
+
+const prettyMS = require('pretty-ms');
+const prettyBytes = require('pretty-bytes');
+const chalk = require('chalk');
+
+morgan.token('pretty-size', (req, res) => {
+  const size = res.get('content-length');
+  if (!size) return;
+  return prettyBytes(Number(size));
+});
+
+morgan.token('pretty-status', (req, res) => {
+  const headersSent =
+    typeof res.headersSent !== 'boolean'
+      ? Boolean(res._header)
+      : res.headersSent;
+  const status = headersSent ? String(res.statusCode) : undefined;
+  if (status >= 500) return chalk.red(status);
+  if (status >= 400) return chalk.yellow(status);
+  return status;
+});
+
+morgan.token('pretty-response-time', (req, res) => {
+  if (!req._startAt || !res._startAt) return;
+  const ms =
+    (res._startAt[0] - req._startAt[0]) * 1e3 +
+    (res._startAt[1] - req._startAt[1]) * 1e-6;
+  return prettyMS(Number(ms.toFixed()));
+});
+
+exports.loggerMiddleware = (logger) =>
+  morgan(':method :url :pretty-status :pretty-response-time - :pretty-size', {
```

Comment:
Should we also have a date / time in here?

Reply:
I do not think so - and both the `tiny` and `dev` default morgan formats have neither.

Reply:
By the, I would only consider this request logging a dev feature, it is not meant for production use. One would probably want to register an additional morgan instance for that.

Reply:
Huh, why don't we use the `dev` preset then? https://github.com/expressjs/morgan#dev

Reply:
Because I wanted to provide properly formatted response sizes..:

https://github.com/untool/untool/blob/f6801345916c386269d11e6c74ae3168eb26fff5/packages/yargs/lib/request.js#L10-L14

...and a single mechanism to manage CLI colors (`chalk` detects Terminal color support and supports `--no-colors` flag IIRC).